### PR TITLE
sct_download_data safety checks

### PR DESCRIPTION
**Pull request by kousu**
Some cleanups I found during #2668, but pulled out separately.

This **depends on** #2700.

### CODE REVIEW

File: scripts/sct_dmri_moco.py [hunk 1]
```diff
@@ -84,8 +84,7 @@ def get_parser():
                                   "poly [int]: Degree of polynomial function used for regularization along Z. For no regularization set to 0. Default=" + param_default.poly + ".\n"
                                                 "smooth [mm]: Smoothing kernel. Default=" + param_default.smooth + ".\n"
                                                   "metric {MI, MeanSquares, CC}: Metric used for registration. Default=" + param_default.metric + ".\n"
-                                                  "gradStep [float]: Searching step used by registration algorithm. The higher the more deformation allowed. Default=" + param_default.gradStep + ".\n"
-                                                    "sample [0-1]: Sampling rate used for registration metric. Default=" + param_default.sampling + ".\n",
+                                                  "gradStep [float]: Searching step used by registration algorithm. The higher the more deformation allowed. Default=" + param_default.gradStep + ".\n",
```
Comment (jcohenadad) [inline scripts/sct_dmri_moco.py:87]: out of PR scope?

Comment (kousu) [reply scripts/sct_dmri_moco.py:87]: This is because I did this off #2642. But I've changed my mind about that. I'm going to do it off #2700 instead.

**Comment by kousu** [comment]
> some of the changes seem out of scope (e.g. moco)-- is that on purpose?

I've rebased onto #2700 instead of leaving this towering on my currently open stack of PRs (#2642, #2666 , #2667 , #2668). It's good to go now and can be merged before them.

File: spinalcordtoolbox/download.py [hunk 2]
```diff
@@ -50,6 +50,9 @@ def download_data(urls):
                 _, content = cgi.parse_header(response.headers['Content-Disposition'])
                 filename = content["filename"]
 
+            # protect against directory traversal
+            filename = os.path.basename(filename)
```
Comment (kousu) [inline spinalcordtoolbox/download.py:55]: Does this need to worry about symlinks?

Comment (jcohenadad) [reply spinalcordtoolbox/download.py:55]: so far we don't have symlinks to download, so it should be fine?

Comment (kousu) [reply spinalcordtoolbox/download.py:55]: This was to protect against [directory traversal](https://owasp.org/www-community/attacks/Path_Traversal).

My thought about symlinks is a related but obscure attack:

1. The mirror -- or someone who manages to impersonate it -- knows that you have a symlink in your download folder, say `keys -> ~/.ssh/authorized_keys`, 
2. They return "Content-Disposition: attachment; filename=keys" with the contents "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIKYKsUxEseqswXVJ3LbmMq6eXkluXNdU6EszFUBx8LUN attacker@evil"
3. They can now ssh into your machine.

**Pull request merged**